### PR TITLE
chore: Automate release on new git tag push

### PR DESCRIPTION
**Pull request by ogenev**
### What was wrong?
Manually adding binaries for a release is a PITA.

### How was it fixed?
Added a git action workflow which should create a new draft release with binaries included on a new git tag push

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history and use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+          mkdir artifacts
+          mv target/${{ matrix.arch }}/${{ matrix.profile }}/trin.exe ./artifacts
+
+      # ==============================
+      #       Signing
+      # ==============================
+
+      - name: Configure GPG and create artifacts
+        env:
+          GPG_SIGNING_KEY: ${{ secrets.GPG_SIGNING_KEY }}
```
Comment (ogenev) [inline .github/workflows/release.yml:105]: @carver do you have admin access to the repo, do you know if we have added `GPG_SIGNING_KEY` and `GPG_PASSPHRASE` as secrets?

**Comment by ogenev** [comment]
I'm unsure how to test this other than merging it and pushing some test tags, so I'm okay with optimistically merging this and iterating if issues pop up.

File: .github/workflows/release.yml (hunk 1 above)
Comment (carver) [reply .github/workflows/release.yml:105]: Looks like no secrets set:
image

Comment (carver) [reply .github/workflows/release.yml:105]: I'm not sure what the standards are for this, but I don't think the answer is for someone to put their private gpg key & passphrase into the repo secrets. I guess we would just generate a new one. If you want to do that and send it to me over keybase, I can save it into those secrets under Actions.

Comment (ogenev) [reply .github/workflows/release.yml:105]: >  If you want to do that and send it to me over keybase, I can save it into those secrets under Actions.

Sent the key over keybase.

**Review by carver — APPROVED** [review]
Frankly, this is kind of a rubber-stamp review. I scanned it but am not so familiar with github actions, so I probably missed it if something is buggy. :)

Testing in prod!

**Comment by carver** [comment]
I'll merge so that we can test on the upcoming deployment.

File: .github/workflows/release.yml (hunk 1 above)
Comment (carver) [reply .github/workflows/release.yml:105]: Saved to the repo 👍🏻

**Comment by carver** [comment]
Hm, no draft release created when I pushed a new tag 🤷🏻‍♂️ -- I'll just build it manually following the previous release flow

**Pull request merged**